### ISSUE TITLE
Adding a Custom class that is same as ModKMeans

### ISSUE THREAD

**Issue by RunKZhang** [vferat/pycrostates#117]
Hey everyone,

I am thinking of adding a new class in the cluster module. In some situations, users may implement new algorithms or methods to do clustering on gfp data rather than using K-means or AAHC. However, there is a lack of corresponding class to integrate the results obtained by new algorithms or methods with the entire pycrostates workflow. 

I am very happy to help improve the pycrostates, and I am working on implementing this custom and general class.

**Comment by mscheltienne** [comment]
Do you mean a `ClusterArray` object, similar to what `RawArray` is for `Raw` in MNE-Python?

**Comment by RunKZhang** [comment]
Yes, something like that but not very similar. Has `ClusterArray` implemented yet?
